### PR TITLE
[Hue] - CDPD-32891 - Cloudbreak Template change for QP

### PR DESCRIPTION
**Pull request by Amlesh1902**
Testing: I have done end-to-end testing and found that it's working as expected.

### CODE REVIEW

File: template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/queryprocessor/QueryProcessorConfigProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    public void getServiceConfigs() {
+        RDSConfig rdsConfig = new RDSConfig();
+        rdsConfig.setType(QUERY_PROCESSOR);
+        rdsConfig.setConnectionURL(String.format("jdbc:%s://%s:%s/%s", DB_PROVIDER, HOST, PORT, DB_NAME));
+        rdsConfig.setConnectionUserName(USER_NAME);
+        rdsConfig.setConnectionPassword(PASSWORD);
+        TemplatePreparationObject tpo = new Builder().withRdsConfigs(Set.of(rdsConfig)).build();
+
+        List<ApiClusterTemplateConfig> result = underTest.getServiceConfigs(null, tpo);
+        Map<String, String> paramToVariable =
```
Comment (lajosrodek) [inline template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/queryprocessor/QueryProcessorConfigProviderTest.java:57]: Nit: Rename to `configToValue`.

File: template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/queryprocessor/QueryProcessorConfigProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+        boolean result = underTest.isConfigurationNeeded(mockTemplateProcessor, tpo);
+        assertThat(result).isFalse();
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void isConfigurationNeededFalseWhenNoDBRegistered() {
+        CmTemplateProcessor mockTemplateProcessor = mock(CmTemplateProcessor.class);
+        when(mockTemplateProcessor.isRoleTypePresentInService(anyString(), any(List.class))).thenReturn(true);
```
Comment (lajosrodek) [inline template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/queryprocessor/QueryProcessorConfigProviderTest.java:115]: Like above.

**Review by lajosrodek — CHANGES_REQUESTED** [review]

File: template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/queryprocessor/QueryProcessorConfigProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+import static org.mockito.Mockito.when;
+
+import java.util.AbstractMap.SimpleEntry;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.junit.Before;
+import org.junit.Test;
```
[the remark this answers is not in the document]
Comment (horadla23) [reply template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/queryprocessor/QueryProcessorConfigProviderTest.java:16]: +1

File: core/src/main/resources/defaults/blueprints/7.2.15/cdp-data-engineering.bp [hunk 4]
```diff
@@ -408,6 +408,17 @@
           }
         ]
       },
+      {
+        "refName": "query_processor",
+        "serviceType": "QUERY_PROCESSOR",
```
[the remark this answers is not in the document]
Comment (Amlesh1902) [reply core/src/main/resources/defaults/blueprints/7.2.15/cdp-data-engineering.bp:413]: 1) "cdp-data-engineering-spark3" - The QP is related to DAS and Hue service, but cdp-data-engineering-spark3 does not have the DAS service. So this is not applicable.
2) "data-engineering-ha.bp" - I will make the change to include QP in this blueprint.

File: template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/queryprocessor/QueryProcessorConfigProviderTest.java (hunk 1 above)
Comment (Amlesh1902) [reply template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/queryprocessor/QueryProcessorConfigProviderTest.java:57]: acknowledged

File: template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/queryprocessor/QueryProcessorConfigProviderTest.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    @Test
+    public void getRoleTypes() {
+        assertThat(underTest.getRoleTypes()).containsOnly(QueryStoreRoles.QUERY_PROCESSOR);
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void isConfigurationNeededTrue() {
+        CmTemplateProcessor mockTemplateProcessor = mock(CmTemplateProcessor.class);
+        when(mockTemplateProcessor.isRoleTypePresentInService(anyString(), any(List.class))).thenReturn(true);
```
[the remark this answers is not in the document]
Comment (Amlesh1902) [reply template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/queryprocessor/QueryProcessorConfigProviderTest.java:81]: acknowledged

File: template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/queryprocessor/QueryProcessorConfigProviderTest.java [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+        boolean result = underTest.isConfigurationNeeded(mockTemplateProcessor, tpo);
+        assertThat(result).isTrue();
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void isConfigurationNeededFalseWhenNoHueQueryProcessorOnClusterr() {
+        CmTemplateProcessor mockTemplateProcessor = mock(CmTemplateProcessor.class);
+        when(mockTemplateProcessor.isRoleTypePresentInService(anyString(), any(List.class))).thenReturn(false);
```
[the remark this answers is not in the document]
Comment (Amlesh1902) [reply template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/queryprocessor/QueryProcessorConfigProviderTest.java:98]: acknowledged

File: template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/queryprocessor/QueryProcessorConfigProviderTest.java (hunk 2 above)
Comment (Amlesh1902) [reply template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/queryprocessor/QueryProcessorConfigProviderTest.java:115]: acknowledged

**Review by lajosrodek — APPROVED** [review]
Thanks, LGTM.

**Comment by biharitomi** [comment]
Please rebase your branch @Amlesh1902 to the latest master.

**Comment by Amlesh1902** [comment]
> Please rebase your branch @Amlesh1902 to the latest master.

Done!

**Comment by lajosrodek** [comment]
@Amlesh1902 Please do not merge `master` into your branch but rather rebase on top of `master`. GitHub still does not allow merging this as of now. Furthermore, the typical preferred approach in this repo is to have only a single commit per PR; this one has 3 due to the 2 extra merge commits. Could you please fix these issues so that this can be finally merged?

**Comment by Amlesh1902** [comment]
> @Amlesh1902 Please do not merge `master` into your branch but rather rebase on top of `master`. GitHub still does not allow merging this as of now. Furthermore, the typical preferred approach in this repo is to have only a single commit per PR; this one has 3 due to the 2 extra merge commits. Could you please fix these issues so that this can be finally merged?

Done!

**Comment by lajosrodek** [comment]
Note: Basically a resubmit of the former #12611.

**Pull request merged**